pkg/bot/lark.go

```diff
@@ -0,0 +1,225 @@
+		if err != nil {
+			log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+			return
+		}
+
+		l.LarkClient.SendTextMessage(ChatID, chatID, response)
+		return
+	}
+	openID, err := AccessAndTypeCastToString(OpenID, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+	l.LarkClient.SendTextMessage(OpenID, openID, response)
+}
+
+// Start starts the lark server and listens for lark messages
+func (l *LarkBot) Start() {
+	// See: https://open.larksuite.com/document/ukTMukTMukTM/ukjNxYjL5YTM24SO2EjN
+	// message
+	larkConf := l.LarkClient.Conf
+	event.SetTypeCallback(larkConf, Message, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.Execute(e)
+		return nil
+	})
+
+	// add_bot
+	event.SetTypeCallback(larkConf, AddBot, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
+		return nil
+	})
+
+	// p2p_chat_create
+	event.SetTypeCallback(larkConf, P2pChatCreate, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
+		return nil
+	})
+
+	// add_user_to_chat
+	event.SetTypeCallback(larkConf, AddUserToChat, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
+		return nil
+	})
+
+	eventhttpserver.Register(l.MessagePath, larkConf)
+	log.Infof("Started lark server on port %d", l.Port)
+	log.Errorf("Error in lark server. %v", http.ListenAndServe(fmt.Sprintf(":%d", l.Port), nil))
+}
+
+// SayHello send welcome message to new added users
+func (l *LarkBot) SayHello(e map[string]interface{}) error {
+	event, err := AccessAndTypeCastToMap(Event, e)
+	if err != nil {
+		return err
+	}
+
+	larkUsers, ok := event[Users]
+	if !ok {
+		return fmt.Errorf("%s not found", larkUsers)
+	}
+	users, ok := larkUsers.([]interface{})
+	if !ok {
+		user, ok := event[Users].(interface{})
+		if !ok {
+			return fmt.Errorf("typecast error - failed to convert into interface{}")
+		}
+		users = append(users, user)
+	}
+
+	var messages []string
+	if users != nil {
+		for _, user := range users {
+			userMap, ok := user.(map[string]interface{})
+			if !ok {
+				log.Error("typecast error - failed to convert into interface{}")
+				continue
+			}
+			openID, err := AccessAndTypeCastToString(OpenID, userMap)
+			if err != nil {
+				log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+				continue
+			}
+			username, err := AccessAndTypeCastToString(UserID, userMap)
+			if err != nil {
+				log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+				continue
+			}
+
+			messages = append(messages, fmt.Sprintf("<at user_id=\"%s\">%s</at>", openID, username))
+		}
+	}
+	messages = append(messages, "Hello from botkube~ Play with me by at botkube <commands>")
```

Comment:
Let's declare the msg as const. The msg can be improved as well. E,g this is the generic welcome msg when BotKube start
https://github.com/infracloudio/botkube/blob/develop/pkg/controller/controller.go#L50